modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts

```diff
@@ -0,0 +1,127 @@
+import { AddEventsBehaviour, AlloyComponent, AlloyEvents, Behaviour, Button, Focusing, Input, Keying, NativeEvents, Representing, SketchSpec } from '@ephox/alloy';
+import { Cell, Fun, Id, Optional } from '@ephox/katamari';
+import { Focus, SugarElement, Traverse } from '@ephox/sugar';
+
+import Editor from 'tinymce/core/api/Editor';
+import { UiFactoryBackstage } from 'tinymce/themes/silver/backstage/Backstage';
+
+import { onControlAttached, onControlDetached } from '../../controls/Controls';
+import { updateMenuText, UpdateMenuTextEvent } from '../../dropdown/CommonDropdown';
+import { onSetupEvent } from '../ControlUtils';
+import { NumberInputSpec } from './FontSizeBespoke';
+
+interface BespokeSelectApi {
+  readonly getComponent: () => AlloyComponent;
+}
+
+const createBespokeNumberInput = (editor: Editor, _backstage: UiFactoryBackstage, spec: NumberInputSpec): SketchSpec => {
```

Comment:
Typically, you would only return the type `SketchSpec` if you were writing a factory function for a sketcher. Otherwise, you'll just return `AlloySpec`. That would also mean you don't have to return a `uid`, which is preferred. It's better if alloy generates them (which it does if they aren't supplied)

Reply:
changed